app/controllers/nicknames.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+const Room = require('../models/room.js');
+const Nickname = require('../models/nickname.js');
+const logger = require('./../logger/winston');
+const channel = logger.init('error');
+
+exports.getNicknameByUserRoom = async (req, res) => {
+  const { roomId } = req.params;
+  const { _id: userId } = req.decoded;
+
+  try {
+    let nicknames = {};
+    const results = await Nickname.getList(userId, roomId);
+
+    if (results.length > 0) {
+      nicknames = results.reduce(function(object, nickname, i) {
```

Comment:
es6 viết lại function cho nó nguy hiểm đi (len)
```suggestion
      nicknames = results.reduce((object, nickname, i) =>  {
```